src/plugins/tool_extrude.py

```diff
@@ -29,128 +29,102 @@ def __init__(self, api):
-            return True
-        return False
-
-    def on_mouse_click(self, data):
-        shift_down = not not data.key_modifiers & QtCore.Qt.KeyboardModifier.ShiftModifier
-        if shift_down and self.regionstart is None:
-            self.regionstart = [data.world_x, data.world_y, data.world_z]
-        elif self.regionstart is not None:
-            self.regionend = [data.world_x, data.world_y, data.world_z]
-            region, valid = self.regionvalid()
-            if valid:
-                if region == 0:
-                    directionHint = "\nPositive values mean to the right, negative mean to the left."
-                    min = -1 * data.world_x
-                    max = data.voxels.width - data.world_x - 1
-                if region == 1:
-                    directionHint = "\nPositive values mean up, negative mean down."
-                    min = -1 * data.world_y
-                    max = data.voxels.height - data.world_y - 1
-                if region == 2:
-                    directionHint = "\nPositive values mean to the back, negative mean to the front."
-                    min = -1 * data.world_z
-                    max = data.voxels.depth - data.world_z - 1
-                value, ret = QtGui.QInputDialog.getInt(QtGui.QApplication.instance().mainwindow, "Extrude",
-                                                       "Extrude region: " + str(self.regionstart) + " - " +
-                                                       str(self.regionend) + directionHint, 0, min, max)
-                if ret:
-                    self.do_extrude(data, value, region)
-            else:
-                QtGui.QMessageBox.warning(QtGui.QApplication.instance().mainwindow, "Extrude",
-                                          ("The region you selected is invalid. Try again.\n"
-                                           "Make sure that they are on a plane in any direction.\n"
-                                           "Extrusion can only be done in straigt directions. "
-                                           "Top, bottom, left, right, back and front."), QtGui.QMessageBox.Ok)
-            self.regionstart = None
-            self.regionend = None
-        else:
-            QtGui.QMessageBox.warning(QtGui.QApplication.instance().mainwindow, "Extrude",
-                                      ("You have not selected a region yet. "
-                                       "Hold shift and click on two voxels that are on a flat plane in any direction."),
-                                      QtGui.QMessageBox.Ok)
+        self._mouse = None
+        self._stamp = []
+        self.xdir = True
+        self.ydir = True
+        self.zdir = True
+        self.pastoffset = 0
+        
 
+    def drawstamp(self, data, dx, dy, dz):
+        for x, y, z, col in self._stamp:
+            tgt = data.voxels.get(x+dx,  y+dy,  z+dz)
+            if tgt == 0:
+                data.voxels.set(x+dx,  y+dy, z+dz, col,  True,  1)
```

Comment:
Could you please follow the [pep8 style guide](https://www.python.org/dev/peps/pep-0008/) now as the rest of the project is converted to follow it as of yesterday.
* use one space around operators
* only one space after `,`